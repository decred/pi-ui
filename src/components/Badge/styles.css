.badgeWrapper {
  position: absolute;
  display: inline-flex;
  align-items: center;
  flex-direction: row;
  padding: 2rem 3rem 2rem 0;
  z-index: 100;
  border-radius: 5px;
  height: 5rem;
  -webkit-box-shadow: 0px 3px 6px 2px rgba(0, 0, 0, 0.16);
  -moz-box-shadow: 0px 3px 6px 2px rgba(0, 0, 0, 0.16);
  box-shadow: 0px 3px 6px 2px rgba(0, 0, 0, 0.16);
  visibility: hidden;
  opacity: 0;
}

.badgeWrapperVisible {
  composes: badgeWrapper;
  visibility: visible;
  opacity: 1;
}

.badgeClose,
.badgeClose:link,
.badgeClose:visited {
  position: absolute;
  top: 0;
  right: 1rem;
  font-size: var(--font-size-xxlarge);
  color: var(--text-color-light);
}

.badgeClose:hover,
.badgeClose:link:hover,
.badgeClose:visited:hover {
  color: var(--color-primary);
}

.badgeIcon {
  display: inline-flex;
  align-items: center;
  flex-direction: row;
  padding: 0 1rem 0 2rem;
}
