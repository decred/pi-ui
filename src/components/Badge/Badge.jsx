import React from "react";
import PropTypes from "prop-types";
import styles from "./styles.css";
import { classNames } from "../../utils";

const Badge = ({ children, icon, show, onClose, ...props }) => {
  const onCloseClick = (e) => {
    e.preventDefault();
    onClose();
  };
  return (
    <div
      className={classNames(
        show ? styles.badgeWrapperVisible : styles.badgeWrapper
      )}
      {...props}>
      <a className={styles.badgeClose} onClick={onCloseClick} href="#">
        &times;
      </a>
      <div className={styles.badgeIcon}>{icon}</div>
      {children}
    </div>
  );
};

Badge.propTypes = {
  children: PropTypes.node.isRequired,
  show: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
  icon: PropTypes.node
};

export default Badge;
