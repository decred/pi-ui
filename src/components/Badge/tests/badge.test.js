import React from "react";
import Badge from "../Badge";
import { create } from "react-test-renderer";
import { fireEvent, render } from "@testing-library/react";

describe("Badge component", () => {
  test("Matches the snapshot", () => {
    const badge = create(
      <Badge show={true} onClose={jest.fn()}>
        test
      </Badge>
    );
    expect(badge.toJSON()).toMatchSnapshot();
  });

  test("Calls onClose() prop on close icon click", () => {
    const mockedOnClose = jest.fn();

    const { getByTestId, queryByTestId } = render(
      <Badge show={true} onClose={mockedOnClose}>
        test
      </Badge>
    );

    // close button rendered
    expect(queryByTestId("close-button")).toBeTruthy();

    // click close button
    fireEvent.click(getByTestId("close-button"));

    // expect onClose to be called
    expect(mockedOnClose).toHaveBeenCalledTimes(1);
  });
});
