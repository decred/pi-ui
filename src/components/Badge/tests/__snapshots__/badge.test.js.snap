// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge component Matches the snapshot 1`] = `
<div
  className="badgeWrapperVisible"
>
  <a
    className="badgeClose"
    data-testid="close-button"
    href="#"
    onClick={[Function]}
  >
    Ã—
  </a>
  <div
    className="badgeIcon"
  />
  test
</div>
`;
